apps/nextjs/src/app/api/chat/chatHandler.ts

```diff
@@ -26,6 +26,41 @@ export async function GET() {
   return new Response("Chat API is working", { status: 200 });
 }
 
+async function authoriseAndGetChat({
```

Comment:
I'm trying to move all of this kind of logic out of the controller and into the Aila library. Could we make this logic happen there instead?